CommonTools/RecoAlgos/src/ClusterStorer.cc

```diff
@@ -182,4 +182,28 @@ namespace helper {
     (*cluRef) = OmniClusterRef(newRef);
   }
 
+  // -------------------------------------------------------------
+  // Specific rekey for class template ClusterRefType = VectorHit::ClusterRef,
+  // RecHitType is not used.
+  template <>
+  template <typename RecHitType>  // or template<> to specialise also here?
+  void ClusterStorer::ClusterHitRecord<VectorHit::ClusterRef>::
+      //  rekey<VectorHit>(const VectorHit::ClusterRef &newRef) const
+      rekey(const VectorHit::ClusterRef &newRef) const {
```

Comment:
seems very strange that a `const` function is modifying the state of the object.

Reply:
it's the same pattern as

https://github.com/cms-sw/cmssw/blob/9607861f39b53236a68cf53c6b8cf64b39473eec/CommonTools/RecoAlgos/src/ClusterStorer.cc#L146 

and

https://github.com/cms-sw/cmssw/blob/9607861f39b53236a68cf53c6b8cf64b39473eec/CommonTools/RecoAlgos/src/ClusterStorer.cc#L164

Reply:
As far as I can tell, the `rekey()` does not have to be `const`. The only place I saw it is called is in `ClusterStorer::processClusters()`
https://github.com/cms-sw/cmssw/blob/9607861f39b53236a68cf53c6b8cf64b39473eec/CommonTools/RecoAlgos/src/ClusterStorer.cc#L133
which is not a `const` method, so it could use a non-`const` `::iterator` instead of
https://github.com/cms-sw/cmssw/blob/9607861f39b53236a68cf53c6b8cf64b39473eec/CommonTools/RecoAlgos/src/ClusterStorer.cc#L108

Reply:
resolved in https://github.com/cms-sw/cmssw/pull/40363/commits/8195ef2e3cf0a1a75ac2486ea373222c9fdc1628

Reply:
Thanks!